### PR TITLE
salief/stale auth

### PR DESCRIPTION
**Pull request by salieflewis**
- **rm auth token instance from user context**
- **invoke get access token within item dropzone comp**
- **refactor api file to invoke get access token in the context of respective fns**

### CODE REVIEW

File: apps/site/lib/api.ts [hunk 1]
```diff
@@ -80,7 +84,8 @@ export async function relayRegisterFor(
 
 /* MEDIA SERVICE */
 
-export async function w3sUpload(body: FormData, authToken: string | null) {
+export async function w3sUpload(body: FormData) {
+  const authToken = await getAccessToken()
```
Comment (0xTranqui) [inline apps/site/lib/api.ts:88]: wonder if theres any reason to block the request from going out if the auth token doesnt come back correctly? even tho i assume its cleaner to have the check be in one place (in this case, the media service ?). not sure tho. not mad at this also

File: apps/site/lib/api.ts [hunk 2]
```diff
@@ -106,23 +113,31 @@ export async function w3sUpload(body: FormData, authToken: string | null) {
...
+      'https://river-media-service.up.railway.app/mux/upload',
+      {
+        method: 'POST',
+        headers: authToken
+          ? { Authorization: `Bearer ${authToken}` }
+          : undefined,
+        body,
+      },
+    )
+    if (!res.ok) {
```
Comment (0xTranqui) [inline apps/site/lib/api.ts:129]: could we potentially log exactly the issue here? which is like "request to mux media service failed?"

**Review by 0xTranqui — APPROVED** [review]
looks good to merge! dropped a few smol comments / qs

File: apps/site/lib/api.ts (hunk 1 above)
Comment (salieflewis) [reply apps/site/lib/api.ts:88]: The media service throws an error if there isn't an `authToken` present in the header of the request. We should ensure that that error gets returned to the client so it is clear why a request failed.

**Pull request merged**